server/resp/src/main/java/org/infinispan/server/resp/commands/ArgumentUtils.java

```diff
@@ -39,26 +39,48 @@ public static double toDouble(byte[] argument, int offset) {
       if (argument == null || argument.length == 0)
          throw new NumberFormatException("Empty argument");
 
-      return toDoubleWithInfinity(new String(argument, offset, argument.length - offset, CharsetUtil.US_ASCII));
+      String s = new String(argument, offset, argument.length - offset, CharsetUtil.US_ASCII);
+      return toDoubleWithInfinity(s, argument, offset);
    }
 
    public static double toDouble(byte[] argument) {
       if (argument == null || argument.length == 0)
          throw new NumberFormatException("Empty argument");
       String sArg = toNumberString(argument);
-      return toDoubleWithInfinity(sArg);
+      return toDoubleWithInfinity(sArg, argument, 0);
    }
 
-   private static double toDoubleWithInfinity(String sArg) {
-      if (sArg.equalsIgnoreCase("Inf") || sArg.equalsIgnoreCase("+Inf")) {
-      sArg="Infinity";
+   private static double toDoubleWithInfinity(String sArg, byte[] number, int offset) {
+      if (sArg.equalsIgnoreCase("Inf")
+            || sArg.equalsIgnoreCase("+Inf")
+            || sArg.equals("Infinity")) {
+         return Double.POSITIVE_INFINITY;
       }
-      if (sArg.equalsIgnoreCase("-Inf")) {
-      sArg="-Infinity";
+
+      if (sArg.equalsIgnoreCase("-Inf") || sArg.equals("-Infinity")) {
+         return Double.NEGATIVE_INFINITY;
       }
+
+      assertNumericString(number, offset);
```

Comment:
why this additional check, isn't `parseDouble` enough?

Reply:
Unfortunately, no. It accepts some inputs that Redis does not accept. For example, `Double.parseDouble("   1.0      ")` is valid. It would be much simpler to call it directly :(.

Reply:
oh ok, thanks for explaining!